core/asm/lexer.go

```diff
@@ -254,7 +254,7 @@ func lexInsideString(l *lexer) stateFn {
 
 func lexNumber(l *lexer) stateFn {
 	acceptance := Numbers
-	if l.accept("0") && l.accept("xX") {
+	if l.accept("0") || l.accept("xX") {
```

Comment:
It seems that the '0' character in "0x" has already been consumed by "lexLine()" function. Thus, the input of "lexNumber()" function only remains 'x' character.